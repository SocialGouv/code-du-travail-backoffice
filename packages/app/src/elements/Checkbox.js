import PropTypes from "prop-types";
import React from "react";

import Button from "./Button";

const Checkbox = ({ isChecked = false, ...props }) => (
  <Button icon={isChecked ? "check-square" : "square"} isTransparent role="checkbox" {...props} />
);

Checkbox.propTypes = {
  ...Button.propTypes,
  isChecked: PropTypes.bool.isRequired,
};

export default Checkbox;
