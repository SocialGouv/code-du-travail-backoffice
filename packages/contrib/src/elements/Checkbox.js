import React from "react";

import Button from "./Button";

const Checkbox = ({ color = "primary", isChecked = false, onClick, ...props }) => (
  <Button
    color={color}
    icon={isChecked ? "check-square" : "square"}
    onClick={onClick}
    role="checkbox"
    {...props}
  />
);

export default Checkbox;
