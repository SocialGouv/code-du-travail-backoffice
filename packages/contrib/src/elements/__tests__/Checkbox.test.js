import React from "react";
import { fireEvent, render } from "@testing-library/react";

import Checkbox from "../Checkbox";

describe("[Contrib] elements/<Checkbox />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Checkbox />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with a custom `color` prop", () => {
    const { container } = render(<Checkbox color="secondary" />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with `isChecked` prop", () => {
    const { container } = render(<Checkbox isChecked />);

    expect(container).toMatchSnapshot();
  });

  it("should have called onClick()", () => {
    const props = {
      "data-testid": generateTestId(),
      onClick: jest.fn()
    };
    const λ = render(<Checkbox {...props} />);

    fireEvent.click(λ.getByTestId(props["data-testid"]));

    expect(props.onClick).toHaveBeenCalled();
  });
});
