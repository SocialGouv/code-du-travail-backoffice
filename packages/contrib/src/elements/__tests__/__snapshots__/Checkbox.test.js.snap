// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<Checkbox /> should match snapshot 1`] = `
<div>
  <button
    class="Checkbox-sc-1jox9lb-0 efNukH Button-sc-1bpx84u-0 eyWUJN css-vhv7r1"
    role="checkbox"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-square fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
      color="var(--color-lapis-lazuli)"
      data-icon="square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`[Contrib] elements/<Checkbox /> should match snapshot with \`isChecked\` prop 1`] = `
<div>
  <button
    class="Checkbox-sc-1jox9lb-0 efNukH Button-sc-1bpx84u-0 eyWUJN css-vhv7r1"
    role="checkbox"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check-square fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
      color="var(--color-lapis-lazuli)"
      data-icon="check-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`[Contrib] elements/<Checkbox /> should match snapshot with a custom \`color\` prop 1`] = `
<div>
  <button
    class="Checkbox-sc-1jox9lb-0 efNukH Button-sc-1bpx84u-0 fKlnvv css-vhv7r1"
    role="checkbox"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-square fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
      color="var(--color-periwinkle)"
      data-icon="square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
